modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -528,6 +528,36 @@ public ObjectDefinition enableAccountEntryRestricted(
 		return objectDefinitionPersistence.update(objectDefinition2);
 	}
 
+	@Override
+	public ObjectDefinition
+			enableAccountEntryRestrictedForNondefaultStorageType(
+				ObjectField objectField)
+		throws PortalException {
+
+		if (!Objects.equals(objectField.getBusinessType(), "Integer") &&
```

Comment:
Hey @guilhermedcamacho,
I don't know if you are aware, but we have a method that compares this [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectFieldImpl.java#L34-L40). I think we could use this method in this condition. Besides that, we already have [constants](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-api/src/main/java/com/liferay/object/constants/ObjectFieldConstants.java) that represent the business field types ("Integer.  "LongInteger", etc)

What do you think about these changes? :smile:

Reply:
I'm adding a label "author action required" because I would like your input on this :+1:

Reply:
Hi @marcelabc , I didn't know that, thanks for let us know. 
cc @nNilton